rails_application/app/controllers/client/orders_controller.rb

```diff
@@ -12,13 +12,12 @@ def new
     end
 
     def create
-      ActiveRecord::Base.transaction do
-        command_bus.(Ordering::SubmitOrder.new(order_id: params[:order_id]))
-        command_bus.(Crm::AssignCustomerToOrder.new(customer_id: cookies[:client_id], order_id: params[:order_id]))
+      Orders::SubmitService.new(order_id: params[:order_id], customer_id: cookies[:client_id]).call.then do |result|
```

Comment:
The code in this controller got more and more complex so I extracted it to a service. 
I used a very simple Service object implementation combined with a simple but flexible Result object. This results in an expressive DSL like the one below :)